numpy/lib/tests/test_io.py

```diff
@@ -1823,6 +1828,123 @@ def test_auto_dtype_largeint(self):
         assert_equal(test['f1'], 17179869184)
         assert_equal(test['f2'], 1024)
 
+
+class TestPathUsage(TestCase):
+    @contextlib.contextmanager
```

Comment:
There are some protections, only the owner can access the file and I would expect other calls to mkstemp would have to return a different name as long as the original exists. There are also suffixes and prefixes that can be specified. The name cannot be reused by another process unless it manages to guess the name, etc, so I expect it is pretty safe. Note some of the usage patterns for mkstemp in the same test file.

For the way forward, probably the easiest thing to do is create a nice context handler in the test file and use that for this PR. Then make another PR moving the manager to `numpy/testing/utils.py` and maybe fixing up some other tests to use it.